From 88c4180ed7211056f58ea8b9610fcf6edbf61089 Mon Sep 17 00:00:00 2001
From: Dinesh Sudham <dineshx.sudham@intel.com>
Date: Sun, 4 Oct 2020 14:34:36 +0800
Subject: [PATCH] DRVLIB_SW-3903:GSWAPI: using GFP_ATOMIC flag allocating
 memory

GFP_KERNEL might block and used for process context, GFP_ATOMIC
doesn't sleep, used for interrupt prcess, bottom halves and cann't
sleep.

Signed-off-by: Dinesh Sudham <dineshx.sudham@intel.com>
---
 drivers/net/ethernet/lantiq/switch-api/gsw_irq.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/lantiq/switch-api/gsw_irq.c b/drivers/net/ethernet/lantiq/switch-api/gsw_irq.c
index 995e65d955da..cdc6838158ee 100644
--- a/drivers/net/ethernet/lantiq/switch-api/gsw_irq.c
+++ b/drivers/net/ethernet/lantiq/switch-api/gsw_irq.c
@@ -113,7 +113,7 @@ static GSW_return_t pce_irq_add(void *cdev, gsw_pce_irq *pce_irq,
 	}
 
 #ifdef __KERNEL__
-	reg_irq = (gsw_pce_irq *)kzalloc(sizeof(gsw_pce_irq), GFP_KERNEL);
+	reg_irq = (gsw_pce_irq *)kzalloc(sizeof(gsw_pce_irq), GFP_ATOMIC);
 	if (!reg_irq) {
 		pr_err("Out of Memory %s:%s:%d",
 		       __FILE__, __func__, __LINE__);
